vars/createSctRunner.groovy

```diff
@@ -15,8 +15,10 @@ def call(Map params, Integer test_duration, String region) {
     set -xe
     env
 
+    # clean the old sct_runner_ip file
+    rm -fv sct_runner_ip
```

Comment:
it is already done in sct.create_runner_instance
![image](https://user-images.githubusercontent.com/5269241/135754132-2253042d-fccf-4afb-ac73-5fdda5ba9579.png)

Reply:
but artifacts pipeline doesn't create `sct_runner_instance`

Reply:
Currently we only try to remove the sct_runner_ip when we want to use the sct runner in the test.
I changed to remove the sct_runner_ip file all the time, even we won't use sct runner in the test.

If there is a old sct_runner_ip file exists, and general runSendEmail(), runCollectLogs(), runCleanupReources() might try to execute on the invalid runner. So it's helpful to remove the old file.

Reply:
The code quoted by @bentsi is in create_runner_instance(). My fix is for the situation that job doesn't create sct runner.

Reply:
1) please don't Resolve comments, only the person who commented can click Resolve (see best practice doc)
2) again, this change is in `vars/createSctRunner.groovy` that runs `./docker/env/hydra.sh create-runner-instance` that triggers `sct.py:create_runner_instance` that has code to remove the old `sct_runner_ip`...

Reply:
@bentsi , artifactsPipeline doesn't call `create-runner-instance`

Reply:
that is why i say the correct fix for this issue is #3994

Reply:
Thanks, changed this thread back to unresolved status.
I understand your description in 2.

Reply:
@amoskong please remove the changes to `vars/createSctRunner.groovy` so we can merge the PR

Reply:
actually we don't need the line `rm -fv sct_runner_ip` at all

Reply:
so you can just remove it

Reply:
I just updated the PR to split it to two patches.
If you still can't agree or understand the second patch, we can only apply the first patch to fix the artifact-test problem.

Reply:
@amoskong we don't run `createSctRunner` in the pipelines that don't need SCT Runner... so this change is not needed. If you think otherwise please point me to the pipeline that uses it but really doesn't need SCT Runner.
I can't merge this change since it does not affect anything

Reply:
We have longevity and upgrade jobs that use `k8s-eks` backend.

The longevity and upgrade pipeline will call createSctRunner(), but sct runner won't be created.

Reply:
My another concern is managerPipeline, I just double check the latest code, it won't be a problem.
If it fails to create the sct runner for the gce region,  the other functions (runCleanupResources, runCollectLogs) will run without runner.

@bentsi  So it's `k8s-eks` backend longevity/upgrade isn't a real issue, we can ignore the patch2.